compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/node/InteropContainer.skiko.kt

```diff
@@ -126,11 +127,21 @@ private const val TRAVERSAL_NODE_KEY =
 internal class TrackInteropModifierNode<T>(
     var container: InteropContainer<T>?,
     var nativeView: T?,
-) : Modifier.Node(), TraversableNode, LayoutAwareModifierNode {
+) : Modifier.Node(), TraversableNode, LayoutAwareModifierNode, OnUnplacedModifierNode {
     override val traverseKey = TRAVERSAL_NODE_KEY
 
     override fun onPlaced(coordinates: LayoutCoordinates) {
         val nativeView = nativeView ?: return
         container?.placeInteropView(nativeView)
     }
+
+    override fun onUnplaced() {
+        val nativeView = nativeView ?: return
+        container?.unplaceInteropView(nativeView)
+    }
+
+    override fun onDetach() {
+        onUnplaced()
```

Comment:
Better to delegate to a new method, which both `onDetach` and `onUnplaced` will call. Otherwise you could end up calling a subclass's `onUnplaced` override, which may be unexpected for it.

Reply:
It's final and not supposed to be subclassed

Reply:
That could change, and even if not, it's cleaner to have two "event listeners" delegate the action to a 3rd method, rather than have one call the other.

Reply:
Disagree. `onUnplaced` is supposed to be called on detaching, it's not separate logic. It's supposed to be called automatically in polished version from AOSP